Klipper_Files/Extra module/ercf.py

```diff
@@ -979,6 +989,49 @@ def _home_to_extruder(self, length, step):
         self._log_info("Failed to reach extruder after moving %.1fmm, pausing" % length)
         self._pause()
 
+    def _home_to_extruder_with_stallguard(self, max_length):
+        self._servo_down()
+        self.toolhead.dwell(0.2)
+        self.toolhead.wait_moves()
+
+        self._log_debug("Homing to extruder with stallguard, up to %.1fmm" % max_length)
+
+        self.gear_stepper.do_set_position(0.)
+
+        pre_move_position = self._counter.get_distance()
+        self.gear_stepper.do_homing_move(max_length, 5, self.gear_stepper_accel, True, True)
+        self.toolhead.dwell(0.2)
+        self.toolhead.wait_moves()
+        post_move_position = self._counter.get_distance()
+        distance_moved = post_move_position - pre_move_position
+        if distance_moved >= max_length:
+            self._log_info("Failed to reach extruder after moving %.1fmm, pausing" % distance_moved)
+            self._pause()
+        else:
+            self._log_debug("Extruder reached after %.1fmm" % distance_moved)
+            return post_move_position
+
+
+    def cmd__ERCF_HOME_FILAMENT_TO_EXTRUDER(self, params):
+        """Test command to home the filament to the extruder from
+           the end of the fast moves down the reverse bowden.
+           
+           Intended to be used for calibrating the stallguard threshold.
+        """
+        return_after = params.get_int('RETURN_AFTER', 0, minval=0, maxval=1)
+        pre_home_position = self._counter.get_distance()
+        if not self.toolhead_use_stallguard:
```

Comment:
Duplicate code!
remove the condition and call _home_to_extruder, this method contains the same condition